src/commands/test/runner.py

```diff
@@ -150,22 +166,45 @@ async def invoke_test_function(self, function_name: str):
         )
 
         func = getattr(self.test_contract, function_name)
+        is_failure_expected = (
+            function_name.startswith("test_fail_")
+            if self._is_test_fail_enabled
+            else False
+        )
         # TODO: Improve stacktrace
         try:
-            call_result = await func().invoke()
-            if self._is_test_error_expected:
-                raise MissingExceptReportedException(
-                    "Expected a transaction to be reverted"
+            try:
```

Comment:
Please move this try into separate method to decrease cognitive complexity

Reply:
If I move it to a separate method, the code will be still hard to understand. I think even more, because
```python
except TestNotFailedReportedException as ex:
            raise ex
```
is justified by the logic of the code inside the inner `try`.

The code is hard to understand because it supports unintuitive use-cases e.g.:
if a test reverts, an exception was expected, and the test was expected to fail, the test will fail.